include/ur_api.h

```diff
@@ -1506,9 +1510,60 @@ urMemImageCreate(
     ur_mem_handle_t *phMem                 ///< [out] pointer to handle of image object created
 );
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Buffer creation properties
+typedef struct ur_buffer_properties_t {
+    ur_structure_type_t stype; ///< [in] type of this structure, must be
+                               ///< ::UR_STRUCTURE_TYPE_BUFFER_PROPERTIES
+    void *pNext;               ///< [in,out][optional] pointer to extension-specific structure
+    void *pHost;               ///< [in][optional] pointer to the buffer data
```

Comment:
pHost is a pointer to user allocated data. I think putting it inside a "properties" struct is slightly confusing. Generally properties are enums or something equivalent. 

I would personally keep it as a separate parameter. However, it could be that I'm biased by the way OpenCL implements this function.